h2o-automl/src/test/java/ai/h2o/automl/TargetEncodingAirlinesBenchmarkTest.java

```diff
@@ -0,0 +1,232 @@
+    parms._stopping_metric = ScoreKeeper.StoppingMetric.AUC;
+    parms._stopping_rounds = 5;
+    parms._ignored_columns = new String[]{"IsDepDelayed_REC", "Origin", "Dest"};
+    GBM job = new GBM(parms);
+    GBMModel gbm = job.trainModel().get();
+
+    Assert.assertTrue(job.isStopped());
+
+    Frame preds = gbm.score(testEncoded);
+    hex.ModelMetricsBinomial mm = ModelMetricsBinomial.make(preds.vec(2), testEncoded.vec(parms._response_column));
+    double auc = mm._auc._auc;
+
+    // Without target encoded Origin column
+    double auc2 = trainDefaultGBM(targetColumnName, tec);
+
+    System.out.println("AUC with encoding:" + auc);
+    System.out.println("AUC without encoding:" + auc2);
+
+    Assert.assertTrue(auc2 < auc);
+  }
+
+  private double trainDefaultGBM(String targetColumnName, TargetEncoder tec) {
+    GBMModel gbm2 = null;
+    Scope.enter();
+    try {
+      Frame airlinesTrainWithTEHDefault = parse_test_file(Key.make("airlines_train_d"), "smalldata/airlines/target_encoding/airlines_train_with_teh.csv");
+      Frame airlinesValidDefault = parse_test_file(Key.make("airlines_valid_d"), "smalldata/airlines/target_encoding/airlines_valid.csv");
+      Frame airlinesTestFrameDefault = parse_test_file(Key.make("airlines_test_d"), "smalldata/airlines/AirlinesTest.csv.zip");
+
+      airlinesTrainWithTEHDefault = tec.ensureTargetColumnIsNumericOrBinaryCategorical(airlinesTrainWithTEHDefault, 10);
+      airlinesValidDefault = tec.ensureTargetColumnIsNumericOrBinaryCategorical(airlinesValidDefault, 10);
+      airlinesTestFrameDefault = tec.ensureTargetColumnIsNumericOrBinaryCategorical(airlinesTestFrameDefault, 10);
+
+      printOutFrameAsTable(airlinesTrainWithTEHDefault, true);
+
+      GBMModel.GBMParameters parms2 = new GBMModel.GBMParameters();
+      parms2._train = airlinesTrainWithTEHDefault._key;
+      parms2._response_column = "IsDepDelayed";
+      parms2._score_tree_interval = 10;
+      parms2._ntrees = 1000;
+      parms2._max_depth = 5;
+      parms2._distribution = DistributionFamily.quasibinomial;
+      parms2._valid = airlinesValidDefault._key;
+      parms2._stopping_tolerance = 0.001;
+      parms2._stopping_metric = ScoreKeeper.StoppingMetric.AUC;
+      parms2._stopping_rounds = 5;
+      parms2._ignored_columns = new String[]{"IsDepDelayed_REC"};
+      GBM job2 = new GBM(parms2);
+      gbm2 = job2.trainModel().get();
+
+      Assert.assertTrue(job2.isStopped());
+
+      Frame preds2 = gbm2.score(airlinesTestFrameDefault);
+
+      hex.ModelMetricsBinomial mm2 = ModelMetricsBinomial.make(preds2.vec(2), airlinesTestFrameDefault.vec(parms2._response_column));
+      double auc2 = mm2._auc._auc;
+      return auc2;
+    } finally {
+      if( gbm2 != null ) {
+        gbm2.delete();
+        gbm2.deleteCrossValidationModels();
+      }
+      Scope.exit();
+    }
+  }
+
+  @After
+  public void afterEach() {
+    System.out.println("After each test we do H2O.STORE.clear() and Vec.ESPC.clear()");
```

Comment:
Each test should handle its own clean-up. This is harmful as it makes detecting leaked keys impossible.

Please go over all the tests and make sure there is no such clean-up.

Reply:
@michalkurka Sure, I've just postponed it because for benchmark tests I think there is no harm to remove it like this. I've tested(covered) everything in unit tests.